editor/inserter/index.js

```diff
@@ -2,52 +2,91 @@
  * External dependencies
  */
 import { connect } from 'react-redux';
+import { bindActionCreators } from 'redux';
 
 /**
  * WordPress dependencies
  */
 import { __ } from '@wordpress/i18n';
 import { Dropdown, IconButton } from '@wordpress/components';
 import { createBlock } from '@wordpress/blocks';
+import { Component } from '@wordpress/element';
 
 /**
  * Internal dependencies
  */
 import InserterMenu from './menu';
 import { getBlockInsertionPoint, getEditorMode } from '../selectors';
-import { insertBlock, hideInsertionPoint } from '../actions';
+import {
+	insertBlock,
+	setBlockInsertionPoint,
+	clearBlockInsertionPoint,
+	hideInsertionPoint,
+} from '../actions';
 
-function Inserter( { position, children, onInsertBlock, insertionPoint } ) {
-	return (
-		<Dropdown
-			className="editor-inserter"
-			position={ position }
-			renderToggle={ ( { onToggle, isOpen } ) => (
-				<IconButton
-					icon="insert"
-					label={ __( 'Insert block' ) }
-					onClick={ onToggle }
-					className="editor-inserter__toggle"
-					aria-haspopup="true"
-					aria-expanded={ isOpen }
-				>
-					{ children }
-				</IconButton>
-			) }
-			renderContent={ ( { onClose } ) => {
-				const onInsert = ( name ) => {
-					onInsertBlock(
-						name,
-						insertionPoint
-					);
+class Inserter extends Component {
```

Comment:
I guess we don't need this to be a class component, but probably the same discussion for/against inline props.

Reply:
I hope we will see the day where there is only one supported way of creating components in React. I feel like some components are converted back and forth whenever requirements change.